client-tools/Shell/ClientFeature.cpp

```diff
@@ -198,6 +198,30 @@ arangosh without connecting to a server.)");
                                    arangodb::options::Flags::OsWindows,
                                    arangodb::options::Flags::Uncommon));
 #endif
+
+  options
+      ->addOption(
+          "--compress-transfer",
+          "Compress data for transport between " + basename + " and server.",
+          new BooleanParameter(&_compressTransfer))
+      .setIntroducedIn(31200)
+      .setLongDescription(R"(This option enables transport compression for data
+received by an ArangoDB server.)");
+
+  options
+      ->addOption("--compress-request-threshold",
+                  "The HTTP request body size from which on requests are "
+                  "transparently compressed when sending them to the server.",
+                  new UInt64Parameter(&_compressRequestThreshold))
```

Comment:
Do we really need options for threshold?
At least we can combine this two.
Threshold == max (default) => no compression
Threshold == 0 => compression enabled, threshold chosen automatically
Any other threshold value => compression enabled, and manually chosen threshold

Reply:
IMHO these are two different things:
1. whether or not client compresses requests based on the size of the uncompressed body
2. whether or not to send the `Accept-Encoding: deflate` along with the request
Both things are orthogonal.
If a request is eligible to be compressed, the server may still respond with an uncompressed response.  And if the request is not eligible to be compressed, the server can still send a compressed response if it is considered useful.
That's why I didn't combine the two options in the first place.
We can combine them into a single option now, but what would be the consequence for when the client should send the `Accept-Encoding: deflate` header with its requests?
1. always send it
2. never send it
3. only send it along requests for which the request body was compressed
This is unclear and thus I would favor a separate config option.